src/sdam/common.ts

```diff
@@ -66,16 +66,16 @@ export interface ClusterTime {
   };
 }
 
-/** Shared function to determine clusterTime for a given topology */
-export function resolveClusterTime(
-  topology: Topology | ClientSession,
+/** Shared function to determine clusterTime for a given topology or session */
+export function _advanceClusterTime(
```

Comment:
Just a thought, we could just keep this without the underscore and put `@internal` in the tsdoc?

Reply:
I don't mind renaming, but this particular suggestion would make the name clash with the actual public method, so maybe we can come up with something else?

Reply:
I think the clash is fine because the public one is a method so it can't be called without a session. But maybe: `advanceClusterTimeFor` to imply its for the topology or session... I mean I'm splitting hairs, I just don't love underscore names but I'm not going to hold us up on this 😆

Reply:
Yeah I understand that it's technically fine to use the same name, I just don't like to as a general practice because it makes it harder to tell at a glance (when doing a codebase wide search, for example) whether or not something is referencing the same thing. I don't have objections to something like `advanceClusterTimeFor` or `advanceClusterTimeOn` etc

Reply:
Both of those sound good to me, I leave the decision to you :)

Reply:
Decided to go with underscore